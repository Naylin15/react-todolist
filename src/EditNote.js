import React from 'react';
import { Checkbox, Grid, Button, Input } from 'semantic-ui-react';

class EditNote extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            value: this.props.note.title,
            noteId: this.props.noteId,
        };
    
        this.handleChange = this.handleChange.bind(this);
    }

    handleChange(event) {
        this.setState({value: event.target.value});
    }
    
    render() {

        return (
            <Grid.Row centered columns={3} id={this.state.noteId}>
                <Grid.Column width={2}>
                    <Checkbox />
                </Grid.Column>
                <Grid.Column>
                    <Input value={this.state.value} onChange={this.handleChange}/>
                </Grid.Column>
                <Grid.Column>
                    <Button className='ui green primary button' onClick={() => this.props.saveChanges(this.state.noteId, this.state.value)}>Save</Button>
                </Grid.Column>
            </Grid.Row>
        )    
    }
}

export default EditNote;