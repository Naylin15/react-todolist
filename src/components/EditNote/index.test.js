import React from 'react';
import renderer from 'react-test-renderer'
import EditNote from './index';

it('renders correctly', () => {
    const tree = renderer
      .create(<EditNote note={{title: 'abc', noteId: 0}}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });