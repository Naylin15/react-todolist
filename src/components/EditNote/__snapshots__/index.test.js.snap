// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="centered three column row"
>
  <div
    className="two wide column"
  >
    <div
      className="ui fitted checkbox"
      onChange={[Function]}
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
    >
      <input
        checked={false}
        className="hidden"
        readOnly={true}
        tabIndex={0}
        type="checkbox"
      />
      <label />
    </div>
  </div>
  <div
    className="column"
  >
    <div
      className="ui input"
    >
      <input
        onChange={[Function]}
        type="text"
        value="abc"
      />
    </div>
  </div>
  <div
    className="column"
  >
    <button
      className="ui button ui green primary button"
      onClick={[Function]}
    >
      Save
    </button>
  </div>
</div>
`;
